app/scripts/migrations/096.ts

```diff
@@ -0,0 +1,102 @@
+import { cloneDeep, fromPairs, map } from 'lodash';
+import { hasProperty, isObject } from '@metamask/utils';
+import { CHAIN_IDS } from '../../../shared/constants/network';
+
+type VersionedData = {
+  meta: { version: number };
+  data: Record<string, any>;
+};
+
+export const version = 96;
+
+/**
+ * This migration will operate the following:
+ *
+ * - Delete `showIncomingTransactions` from `featureFlags` in PreferencesController
+ * - Create a new object under PreferencesController named as `incomingTransactionsPreferences`
+ * 1. which will collect all added networks including localhost
+ * 2. then append the test networks
+ * 3. each of them would become a key coming with the value Ture/False from `showIncomingTransactions`
+ *
+ * @param originalVersionedData - Versioned MetaMask extension state, exactly what we persist to dist.
+ * @param originalVersionedData.meta - State metadata.
+ * @param originalVersionedData.meta.version - The current state version.
+ * @param originalVersionedData.data - The persisted MetaMask state, keyed by controller.
+ * @returns Updated versioned MetaMask extension state.
+ */
+export async function migrate(
+  originalVersionedData: VersionedData,
+): Promise<VersionedData> {
+  const versionedData = cloneDeep(originalVersionedData);
+  versionedData.meta.version = version;
+  transformState(versionedData.data);
+  return versionedData;
+}
+
+interface NetworkConfiguration {
+  chainId: Record<string, any>;
+}
+
+interface FeatureFlags {
+  showIncomingTransactions: boolean;
+}
+
+interface IncomingTransactionPreferences {
+  [key: string]: boolean;
+}
+
+interface State {
+  PreferencesController?: {
+    featureFlags?: FeatureFlags;
+    incomingTransactionsPreferences?: IncomingTransactionPreferences;
+  };
+  NetworkController?: {
+    networkConfigurations?: Record<string, NetworkConfiguration>;
+  };
+}
+
+function transformState(state: State) {
```

Comment:
Could you update this to use `Record<string, unknown>` type instead? The use of this type in other migrations was intentional, as it forces us to make zero assumptions about the state shape so that we don't mishandle corrupted state or create corrupted state.

Reply:
Yes absolutely ! will change here [51f8dda9f2911d242e38a9dc05ad623f570fcd18](https://github.com/MetaMask/metamask-extension/pull/20363/commits/51f8dda9f2911d242e38a9dc05ad623f570fcd18)